core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -108,13 +163,63 @@ class InvokerPool(
         }.map { case (name, state) => s"$name: $state" }
         logging.info(this, s"invoker status changed to ${pretty.mkString(", ")}")
     }
+
+    /** Receive activeAcks from Invoker. */
+    activeAckConsumer.onMessage((topic, _, _, bytes) => {
+        val raw = new String(bytes, StandardCharsets.UTF_8)
+        CompletionMessage.parse(raw) match {
+            case Success(m: CompletionMessage) => {
+                self ! InvocationFinishedMessage(m.invoker, m.response.response.statusCode <= 2)
+                activeAckCallback(m)
+            }
+            case Failure(t) => logging.error(this, s"failed processing message: $raw with $t")
+        }
+    })
+
+    /** Receive Ping messages from invokers. */
+    pingConsumer.onMessage((topic, _, _, bytes) => {
+        val raw = new String(bytes, StandardCharsets.UTF_8)
+        PingMessage.parse(raw) match {
+            case Success(p: PingMessage) => self ! p
+            case Failure(t)              => logging.error(this, s"failed processing message: $raw with $t")
+        }
+    })
 }
 
 object InvokerPool {
     def props(
         f: (ActorRefFactory, String) => ActorRef,
         kv: KeyValueStore,
-        cb: String => Unit) = Props(new InvokerPool(f, kv, cb))
+        cb: String => Unit,
+        p: MessageProducer,
+        ackC: MessageConsumer,
+        acb: CompletionMessage => Unit,
+        pc: MessageConsumer) = Props(new InvokerPool(f, kv, cb, p, ackC, acb, pc))
+
+    val testAction = ExecManifest.runtimesManifest.resolveDefaultRuntime("nodejs:6").map { manifest =>
+        new WhiskAction(
+            namespace = EntityPath("whisk.system"),
+            name = EntityName("invokerHealthTestAction"),
+            exec = new CodeExecAsString(manifest, """function main(params) { return params || {}; }""", None))
+    }
+
+    /** Create an test action on startup if it is not there. */
+    def createTestActionForInvokerHealth(db: EntityStore)(implicit logging: Logging, ec: ExecutionContext): Future[WhiskAction] = {
+        implicit val tid = TransactionId.loadbalancer
+        testAction match {
+            case Some(action) =>
+                WhiskAction.get(db, action.docid).recover {
+                    case _: NoDocumentException => WhiskAction.put(db, action)
```

Comment:
I did a little code change:
the action is now created/updated on every startup of the controller.
It will be deleted never.

Now the startup of the controller fails, if the creation f the action is not possible.